Libraries/xcsdk/Sources/SDK/Manager.cpp

```diff
@@ -123,6 +123,27 @@ executablePaths() const
     };
 }
 
+std::vector<std::string> Manager::
+executablePaths(std::vector<xcsdk::SDK::Toolchain::shared_ptr> &toolchains, xcsdk::SDK::Target::shared_ptr &target) const
+{
+    std::vector<std::string> paths;
+
+    for (Toolchain::shared_ptr const &toolchain : toolchains) {
+        std::vector<std::string> toolchainPaths = toolchain->executablePaths();
+        paths.insert(paths.end(), toolchainPaths.begin(), toolchainPaths.end());
+    }
+
+    if (target != nullptr) {
+        std::vector<std::string> targetPaths = target->executablePaths();
```

Comment:
Yeah, it is weird... not sure of the best structure for this. You want to be able to get the target's paths with the target's toolchains, the target's paths with custom toolchains, toolchain paths with no target, and all of them with the right ordering and including the manager's paths... which leads to what you did here. So maybe that is the best after all.